src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -120,16 +121,38 @@ public async Task<AuthenticationResult> RunAsync(CancellationToken cancellationT
             }
         }
 
-        private void LogErrorTelemetryToClient(string errorCode, MsalTelemetryEventDetails telemetryEventDetails, ITelemetryClient[] telemetryClients)
+        private void LogMsalErrorTelemetryToClient(MsalException ex, MsalTelemetryEventDetails telemetryEventDetails, ITelemetryClient[] telemetryClients)
+        {
+            if (telemetryClients.HasEnabledClients(TelemetryConstants.AcquireTokenEventName))
+            {
+                if (ex is MsalClientException clientException)
```

Comment:
For all other exceptions, we should still log the exception type and message

Reply:
They are logged in a separate method. My suggestion is to combine in one method.